examples/interactive_local_hyperdrive_example.py

```diff
@@ -1,118 +1,228 @@
-# Names are reflected on output data frames and plots later
-agent0 = hyperdrive.init_agent(base=FixedPoint(100000), eth=FixedPoint(100), name="alice")
-hyperdrive_agent1 = hyperdrive_2.init_agent(base=FixedPoint(100000), eth=FixedPoint(100), name="bob")
-# Omission of name defaults to wallet address
-hyperdrive_agent2 = hyperdrive.init_agent(base=FixedPoint(100000), eth=FixedPoint(10))
 
-# Add funds to an agent
-agent0.add_funds(base=FixedPoint(100000), eth=FixedPoint(100))
+# We also add an active pool to this agent as well.
+# This allows us to mint base in initialization
+agent2 = chain.init_agent(base=FixedPoint(100_000), eth=FixedPoint(10), pool=hyperdrive0)
+
+# Add funds to an agent for various pools
+agent0.add_funds(base=FixedPoint(1_000_000), pool=hyperdrive0)
+agent1.add_funds(base=FixedPoint(100_000), pool=hyperdrive0)
+
+
+# %% [markdown]
+#####################
+# Executing Trades
+#####################
 
 # %%
-# Here, we execute a trade, where it's calling agent0 + gather data from data pipeline
-# under the hood to allow for error handling and data management
+# Here, we execute a trade on a pool.
 # Return values here mirror the various events emitted from these contract calls
-open_long_event_1 = agent0.open_long(base=FixedPoint(11111))
-open_long_event_1  # pyright: ignore
+# Here, base is unitless and is dependent on the underlying tokens the pool uses.
+open_long_event_1 = agent0.open_long(base=FixedPoint(11111), pool=hyperdrive0)
+
+# %%
+# We can also set an active pool an agent is using to avoid passing in the pool
+# for every trade.
+agent1.set_active(pool=hyperdrive0)
+open_long_event_2 = agent1.open_long(base=FixedPoint(22222))
+
+# Close previous longs from events
+close_long_event_1 = agent0.close_long(
+    maturity_time=open_long_event_1.maturity_time,
+    bonds=open_long_event_1.bond_amount,
+    pool=hyperdrive0,
+)
+
+# Close previous longs from the wallet
+# This wallet is a wallet object that contains positions relevant to a specific pool.
+# This takes a snapshot of the current wallet positions.
+agent0.get_wallet(pool=hyperdrive0)
+# agent1's active pool is set, so no need to pass in a pool.
+agent1.get_wallet()
+# We also provide helper functions to get specific position types.
+agent1_longs = agent1.get_longs()
+close_long_event_2 = agent1.close_long(maturity_time=agent1_longs[0].maturity_time, bonds=agent1_longs[0].balance)
+
+# Shorts
+open_short_event = agent1.open_short(bonds=FixedPoint(33333))
+agent1_shorts = agent1.get_shorts()
+close_short_event = agent1.close_short(maturity_time=agent1_shorts[0].maturity_time, bonds=agent1_shorts[0].balance)
+
+# LP
+add_lp_event = agent1.add_liquidity(base=FixedPoint(44444))
+remove_lp_event = agent1.remove_liquidity(shares=agent1.get_lp())
+
+# The above trades doesn't result in withdraw shares, but the function below allows you
+# to withdrawal shares from the pool.
+# withdraw_shares_event = agent1.redeem_withdraw_share(shares=agent1.get_withdraw_shares())
```

Comment:
would be nice if a user can call this (or an equivalent close_long or close_short) event and it would provide an informative error "Agent has no withdrawal shares; nothing to redeem".  Then we can leave this line un-commented and also it seems like a common thing to do in a loop and ignore the error.

Reply:
would be even better if we threw a custom `InsufficientBalanceError` that the user could catch!

Reply:
This currently throws an "Can't call this function with an argument of 0"

Reply:
hmm ok yeah I guess that's fine for now